src/osd/OSDMap.h

```diff
@@ -174,6 +174,10 @@ class OSDMap {
     float new_full_ratio = -1;
 
     int8_t new_require_min_compat_client = -1;
+    vector<int32_t> new_noout_osds;
```

Comment:
Use a bit vector/array for that, so each osd can be mapped to single bit and located in constant time *and* take less memory at the same time.